src/vmspawn/vmspawn.c

```diff
@@ -2026,10 +2058,20 @@ static int run_virtual_machine(int kvm_device_fd, int vhost_device_fd) {
                         return log_error_errno(r, "Failed to setup event loop to handle VSOCK notify events: %m");
         }
 
-        /* shutdown qemu when we are shutdown */
-        (void) sd_event_add_signal(event, NULL, SIGINT | SD_EVENT_SIGNAL_PROCMASK, on_orderly_shutdown, &child_pidref);
-        (void) sd_event_add_signal(event, NULL, SIGTERM | SD_EVENT_SIGNAL_PROCMASK, on_orderly_shutdown, &child_pidref);
+        /* handle signals from machined accordingly */
+        if (child_cid != VMADDR_CID_ANY && ssh_private_key_path) {
+                ssh_info = (SSHInfo) {
+                        .cid = child_cid,
+                        .private_key_path = ssh_private_key_path,
+                        .port = 22,
+                };
 
+                (void) sd_event_add_signal(event, NULL, SIGINT | SD_EVENT_SIGNAL_PROCMASK, request_reboot, &ssh_info);
```

Comment:
It seems wrong to handle SIGINT as reboot in vmspawn itself. machined is documented to send SIGINT to the container's init process, but that is very different from the container manager. I don't think handling SIGINT in vmspawn as a workaround is going to work.

Reply:
I think this is possible using GetUnitByPID(1) then calling KillUnit(<unit of pid 1>, "leader", <signal sent to vmspawn>)

Reply:
So I have a pushed a new version which behaves like this, I couldn't get mkosi to build an image in time though so please test this before merging.